models/nft/ethereum/nft_ethereum_native_mints.sql

```diff
@@ -0,0 +1,201 @@
+  evt_index,
+  case
+    -- if only one item type was traded: use it's id
+	when size(ids) = 1 then ids[0]
+    -- otherwise there's no meaningful id to display
+	when size(ids) > 1 then ''
+	else null
+  end as tokenId,
+  from,
+  to,
+  contract_address,
+  'erc1155' as token_standard,
+  case
+	when size(ids) > 1 then 'Bundle Trade'
+	when size(ids) = 1 then 'Single Item Trade'
+	else null
+  end as trade_type,
+  -- take the number of unique items as number_of_items
+  -- -- example:
+  -- -- --   ids = ["id_1", "id_2", "id_3"], values = ["1", "1", "2"] => number_of_items = 3
+  -- technically, it should be the sum of values (e.g. 4 in the example),
+  -- but in almost all erc1155 bundle transactions only one of each item type is transferred
+  -- so this implementation almost always be correct 
+  size(values) as number_of_items
+from {{ source('erc1155_ethereum','evt_transferbatch') }}
+where from = '0x0000000000000000000000000000000000000000'
+),
+results_with_possible_duplicates as (
+select
+  'ethereum' as blockchain,
+  ec.namespace as project,
+  '' as version,
+  ercs.evt_block_time as block_time,
+  ercs.tokenId as token_id,
+  tokens_nft.name as collection,
+  prc.price * tx.value / sum(
+    case
+        when ercs.`from` = '0x0000000000000000000000000000000000000000' then 1
+        else 0
+    end
+  ) over (
+    partition by
+      ercs.contract_address,
+      ercs.evt_tx_hash
+  ) / power(10, prc.decimals) as amount_usd,
+  token_standard,
+  trade_type,
+  sum(
+    case
+        when ercs.`from` = '0x0000000000000000000000000000000000000000' then 1
+        else 0
+    end
+  ) over (
+    partition by
+      ercs.contract_address,
+      ercs.evt_tx_hash
+  ) as number_of_items,
+  'Buy' as trade_category,
+  'Mint' as evt_type,
+  ercs.`from` as seller,
+  ercs.`to` as buyer,
+  tx.value / sum(
+    case
+        when ercs.`from` = '0x0000000000000000000000000000000000000000' then 1
+        else 0
+    end
+  ) over (
+    partition by
+      ercs.contract_address,
+      ercs.evt_tx_hash
+  ) / power(10, prc.decimals) as amount_original,
+  tx.value / sum(
+    case
+        when ercs.`from` = '0x0000000000000000000000000000000000000000' then 1
+        else 0
+    end
+  ) over (
+    partition by
+      ercs.contract_address,
+      ercs.evt_tx_hash
+  )  as amount_raw,
+  'ETH' as currency_symbol,
+  '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' as currency_contract,
+  ercs.contract_address as nft_contract_address,
+  ercs.to as project_contract_address,
+  '' as aggregator_name,
+  '' as aggregator_address,
+  ercs.evt_tx_hash as tx_hash,
+  ercs.evt_block_number as block_number,
+  ercs.`from` as tx_from,
+  ercs.`to` as tx_to,
+  ercs.contract_address || '-' || ercs.evt_tx_hash || '-' || coalesce(ercs.tokenId, '') || '-' || ercs.`from` || '-' || ercs.evt_index
+  as unique_trade_id,
+  -- Temporary fix: {{ source('ethereum','contracts') }} can contain duplicates, which would result in duplicate mints.
```

Comment:
This only affects ~20 mints with total value ~$150, so I thought it's negligible. See https://dune.com/queries/1433939.
Also, we can take `first` because namespace (the only column we're taking from `ethereum.contracts`) is identical partitioned by address.

But yes, this could be a source of bugs in the future. So I'll dive into `ethereum.contracts` to find out why there are duplicates. 🕵️

Reply:
@soispoke I confirmed with boxer in the office hours this is a bug in `ethereum.contracts`, meaning we have to either include a temp fix or wait until `ethereum.contracts` is fixed.

Because this only affects ~20 rows with total value  < $150 and the fix is trivial (and thus can easily be removed by anyone when no longer necessary), I'd say we include the temp fix and ship now.

What do you think?